### PR TITLE
Capture an assert failure from cuvs cagra and rephrase error message to include relevant params

### PR DESCRIPTION
**Pull request by lijinf2**
Error message: "k must be less than topk::kMaxCapacity (256)"
Captured with rephrased to f"cagra with ivf_pq build_algo expects intermediate_graph_degree ({intermediate_graph_degree}) to be smaller than 256"

Also added test cases of various type (float64, int16, int32).

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/knn.py [hunk 1]
```diff
@@ -1508,7 +1514,27 @@ def _transform_internal(
                 if isinstance(item, np.ndarray):
                     item = cp.array(item, dtype="float32")
 
-                index_obj = nn_object.build(build_params, item)
+                try:
+                    index_obj = nn_object.build(build_params, item)
+                except Exception as e:
+                    if "k must be less than topk::kMaxCapacity (256)" in str(e):
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/knn.py:1520]: Interesting way to do it.  What is advantage over  checking the error condition before invoking nn_object.build?

Comment (lijinf2) [reply python/src/spark_rapids_ml/knn.py:1520]: topk::kMaxCapacity (currently 256) is a global variable which may be updated to a higher or lower value some day. Hopefully we can catch the potential update by checking the error message after nn_object.build.

**Review by eordentlich — COMMENTED** [review]
👍

**Review by eordentlich — APPROVED** [review]

**Pull request merged**